proposals/params-collections.md

```diff
@@ -0,0 +1,171 @@
+convenience when calling APIs that take other collection types. For example, an `ImmutableArray<T>`, `ReadOnlySpan<T>`, or 
+plain `IEnumerable`. Especially in cases when compiler is able to avoid an implicit array allocation for the purpose of
+creating the collection (`ImmutableArray<T>`, `ReadOnlySpan<T>`, etc). Today, in situations when an API takes a collection type,
+developers add `params` overload that takes an array, construct the target collection and call the original overload with that
+collection, thus consumers of the API have to trade an extra array allocation for convenience.
+
+## Detailed design
+
+### Method parameters
+
+The [Method parameters](https://github.com/dotnet/csharpstandard/blob/draft-v9/standard/classes.md#1562-method-parameters) section is adjusted as follows.
+
+```diff ANTLR
+formal_parameter_list
+    : fixed_parameters
+-    | fixed_parameters ',' parameter_array
++    | fixed_parameters ',' parameter_collection
+-    | parameter_array
++    | parameter_collection
+    ;
+
+-parameter_array
++parameter_collection
+-    : attributes? 'params' array_type identifier
++    : attributes? 'params' 'scoped'? type identifier
+    ;
+```
+
+A *parameter_collection* consists of an optional set of *attributes*, a `params` modifier, an optional `scoped` modifier,
+a *type*, and an *identifier*. A parameter collection declares a single parameter of the given type with the given name.
+The *type* of a parameter collection shall be one of the following valid target types for a collection expression
+(see https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/collection-expressions.md#conversions):
+- A single dimensional *array type* `T[]`
+- A *span type* `System.Span<T>` or `System.ReadOnlySpan<T>`
+- A *type* with a valid [create method](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/collection-expressions.md#create-methods)
+  with *parameter type* `System.ReadOnlySpan<T>` and the method is at least as accessible as the *type*.
+- A *struct* or *class type* that implements `System.Collections.Generic.IEnumerable<T>`
+- A *struct* or *class type* that implements `System.Collections.IEnumerable` and *does not implement* `System.Collections.Generic.IEnumerable<T>`.
+- An *interface type* `System.Collections.Generic.IEnumerable<T>`, `System.Collections.Generic.IReadOnlyCollection<T>`,
+  `System.Collections.Generic.IReadOnlyList<T>`, `System.Collections.Generic.ICollection<T>`, or `System.Collections.Generic.IList<T>`
+
+In a method invocation, a parameter collection permits either a single argument of the given parameter type to be specified, or
+it permits zero or more arguments of the collection [iteration type](https://github.com/dotnet/csharpstandard/blob/draft-v9/standard/statements.md#1395-the-foreach-statement)
+to be specified. Parameter collections are described further in *[Parameter collections](#parameter-collections)*.
```

Comment:
Or it permits a single argument of the parameter collection type.

Reply:
> Or it permits a single argument of the parameter collection type.

Doesn't "a parameter collection permits either a single argument of the ..." part capture that?

Reply:
You're right, missed that.